src/v/transform/logging/event.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "base/seastarx.h"
+#include "model/transform.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+#pragma once
+
+namespace transform::logging {
+using transform_log_level
+  = named_type<ss::log_level, struct transform_log_level_tag>;
+
+uint32_t transform_log_level_to_severity(transform_log_level lvl);
+
+struct transform_log_event {
+    using clock_type = std::chrono::system_clock;
+    transform_log_event() = delete;
+    explicit transform_log_event(
+      model::node_id source,
+      clock_type::time_point ts,
+      ss::log_level level,
+      ss::sstring message)
+      : source_id(source)
+      , ts(ts)
+      , level(level)
+      , message(std::move(message)) {}
+
+    model::node_id source_id;
+    clock_type::time_point ts;
+    transform_log_level level;
+    ss::sstring message;
+
+    friend bool
+    operator==(const transform_log_event&, const transform_log_event&)
+      = default;
+
+    ss::sstring to_json(std::string_view name) const;
```

Comment:
Should this be something like: `void to_json(iobuf&);` so we append to the iobuf directly and don't have duplicate copies anywhere?

Also why is the `name` here?

Reply:
>void `to_json(iobuf&)`

Good idea.

>why is the `name` here?

To ovoid duplicating it on every event.

Reply:
>to avoid duplicating [the transform name] on every event

The material value of this is probably debatable, but it feels baseline somewhat gross to copy the same string into the buffers on every enqueue. I'm ambivalent; the interface cruft is also somewhat gross (though mostly hidden).